src/main/java/Nano.java

```diff
@@ -0,0 +1,178 @@
+import java.util.Scanner;
+
+public class Nano {
+
+    private final static int HORIZONTAL_LINE_LENGTH = 100;
+    private final static int MAX_TASK_COUNT = 100;
+    private final static String NANO_LOGO = "| \\  ||   / \\   | \\  |||  __  |\n"
+            + "||\\\\ ||  / _ \\  ||\\\\ ||| |  | |\n"
+            + "|| \\\\|| //   \\\\ || \\\\||| |__| |\n"
+            + "||  \\_|//     \\\\||  \\_||______|\n";
+    private final static int COMMAND_INDEX = 0;
+    private final static int TASK_NAME_INDEX = 1;
+    public static final String TASK_COMPLETED_MARK = "[x] ";
+    public static final String TASK_UNCOMPLETED_MARK = "[] ";
+    public static final String MESSAGE_TASK_NOT_FOUND = "Task not found.";
+    public static final String MESSAGE_TASK_MARKED = "Great job completing ";
+    public static final String MESSAGE_TASK_ALREADY_DONE = " is already done";
+    public static final String MESSAGE_TASK_UNMARKED = " is marked as undone";
+    private static Task[] tasks;
+    public static void main(String[] args) {
+        //chatbot startup
+        displayWelcomeMessage();
+        initialiseTaskList();
+
+        //receive commands
+        while (true) {
+            String userInput = getUserInput();
+            executeCommand(userInput);
+        }
+    }
+
+    private static void executeCommand(String userInput) {
+        String[] userInputs = processInput(userInput);
+
+        switch (userInputs[COMMAND_INDEX]) {
+        case "list":
+            displayTaskList();
+            break;
+        case "add":
+            addTask(userInputs[TASK_NAME_INDEX]);
+            break;
+        case "mark":
+            markTask(userInputs[TASK_NAME_INDEX]);
+            break;
+        case "unmark":
+            unmarkTask(userInputs[TASK_NAME_INDEX]);
+            break;
+        case "exit":
+            displayExitMessage();
+            System.exit(0);
+        default:
+            displayHelpMessage();
+            break;
+        }
+    }
+
+    private static void unmarkTask(String taskName) {
+        int taskIndex = getTaskIndex(taskName);
+        if (taskIndex == -1) {
+            System.out.println(MESSAGE_TASK_NOT_FOUND);
+        } else {
+            if (tasks[taskIndex].getTaskCompletionStatus()) {
+                tasks[taskIndex].incompleteTask();
+                System.out.println(taskName + MESSAGE_TASK_UNMARKED);
+            } else {
+                System.out.println(taskName + MESSAGE_TASK_ALREADY_DONE);
+            }
+        }
+    }
```

Comment:
The heavy nesting here can be hard for others to understand the code logic. Apart from reducing the nesting level, perhaps this can be improved using better method naming for the Task class and also replacing the magic number "-1" with a constant variable name.